crates/bifrost/src/providers/replicated_loglet/remote_sequencer.rs

```diff
@@ -0,0 +1,554 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::{ops::Deref, sync::Arc};
+
+use tokio::sync::{mpsc, Mutex, OwnedSemaphorePermit, Semaphore};
+
+use restate_core::{
+    network::{
+        rpc_router::{RpcRouter, RpcToken},
+        NetworkError, NetworkSendError, Networking, Outgoing, TransportConnect, WeakConnection,
+    },
+    task_center, ShutdownError, TaskKind,
+};
+use restate_types::{
+    config::Configuration,
+    errors::MaybeRetryableError,
+    logs::{metadata::SegmentIndex, LogId, Record},
+    net::replicated_loglet::{Append, Appended, CommonRequestHeader, SequencerStatus},
+    replicated_loglet::ReplicatedLogletParams,
+    GenerationalNodeId,
+};
+
+use super::rpc_routers::SequencersRpc;
+use crate::loglet::{
+    util::TailOffsetWatch, AppendError, LogletCommit, LogletCommitResolver, OperationError,
+};
+
+/// RemoteSequencer
```

Comment:
I am wondering if a `RemoteSequencer` is the correct name. Maybe we should rename it to SequencerClient? but it can/will do more than being a client to the sequencer.